Expensify/App
# Allow menus to be dynamically positioned.

Please review @roryabraham

### Details
1. Added a new prop `onDimensionsChange` to `WithWindowDimensions` which allows attaching change Listener.
2. Debounced the Dimensions change event on `WithWindowDimensions` by 200ms.
2. Using the above, set the recalculation of the Call menu and Emoji Picker.

### Fixed Issues

Fixes #2517

### Tests / QA Steps
1. Open Expensify.cash on Web or Desktop
2. Log into your account
3. Click into a chat with another user
4. Click the telephone_receiver phone icon in the top-right, and see the Call Option menu
5. Click on any edge of the window and change the size of the window. See that the Call Options menu dynamically moving with the size of the window.
4. Click the Emoji picker icon on the report composer.
5. Click on any edge of the window and change the size of the window. See that the Emoji picker menu dynamically moving with the size of the window.

### Tested On


### Screenshots


#### Web / Desktop


https://user-images.githubusercontent.com/24370807/117215199-872aa100-ae1b-11eb-8af1-54581936bc30.mp4
